.github/workflows/hardware-smoke.yml

```diff
@@ -254,3 +254,212 @@ jobs:
+            ci/tt-metal/build_Release/lib
+            ci/tt-metal/python_env
+            ci/tt-metal/runtime
+          key: metal-build-cache-${{ hashFiles('tt-zephyr-platforms/submanifests/ci.yaml') }}
+
+      - uses: ./tt-zephyr-platforms/.github/workflows/prepare-zephyr
+        if: ${{ steps.metal-cache-restore.outputs.cache-hit != 'true' }}
+        with:
+          app-path: tt-zephyr-platforms
+
+      - name: Fix APT URLs
+        if: ${{ steps.metal-cache-restore.outputs.cache-hit != 'true' }}
+        run: |
+          # Patch APT URLs to use https
+          sed -i -e 's/http/https/g' /etc/apt/sources.list
+          apt update
+
+      - name: Install clang 17
+        if: ${{ steps.metal-cache-restore.outputs.cache-hit != 'true' }}
+        run: |
+          wget https://apt.llvm.org/llvm.sh
+          sed -i "s/http:\/\/apt.llvm.org/https:\/\/apt.llvm.org/" -i llvm.sh
+          chmod +x llvm.sh
+          ./llvm.sh 17 all
+          rm -f llvm.sh
+
+      - name: Clone Metal and Install Dependencies
+        if: ${{ steps.metal-cache-restore.outputs.cache-hit != 'true' }}
+        shell: bash
+        run: |
+          west config manifest.group-filter -- +ci
+          west update
+          cd ci/tt-metal
+          sed -i "s/http:\/\/apt.llvm.org/https:\/\/apt.llvm.org/" -i install_dependencies.sh
+          ./install_dependencies.sh --mode build --docker
+
+      - name: Build Metal Tests
+        if: ${{ steps.metal-cache-restore.outputs.cache-hit != 'true' }}
+        shell: bash
+        run: |
+          # Install Metal Build Dependencies
+          cd ci/tt-metal
+          PYTHON_CMD=python3.10 source ./create_venv.sh
+          ./build_metal.sh --build-tests --build-programming-examples
+
+      - name: Save Metal Build Cache
+        if: ${{ steps.metal-cache-restore.outputs.cache-hit != 'true' }}
+        uses: actions/cache/save@v4
+        with:
+          path: |
+            ci/tt-metal/build_Release/programming_examples
+            ci/tt-metal/build_Release/test
+            ci/tt-metal/build_Release/lib
+            ci/tt-metal/python_env
+            ci/tt-metal/runtime
+          key: metal-build-cache-${{ hashFiles('tt-zephyr-platforms/submanifests/ci.yaml') }}
+
+  metal-smoke-test:
+    needs: [build-metal, smoke-e2e-test]
+    strategy:
+      fail-fast: false
+      matrix:
+        config:
+          - board: p100
+            runs-on:
+              - yyz-zephyr-lab-p100
+          - board: p100a
+            runs-on:
+              - yyz-zephyr-lab-p100a
+    runs-on: ${{ matrix.config.runs-on }}
+    env:
+      "ZEPHYR_SDK_INSTALL_DIR": /opt/toolchains
+    container:
+      image: ghcr.io/zephyrproject-rtos/ci-repo-cache:v0.27.4.20241026
+      volumes:
+        - /dev/hugepages-1G:/dev/hugepages-1G
+        - /opt/tenstorrent/:/opt/tenstorrent/
+      options: '--device /dev/tenstorrent --device /dev/bus/usb --privileged'
+    steps:
+      - uses: actions/checkout@v4
```

Comment:
could add a `- uses: build-metal.yml` or something along those lines if the above were a reusable workflow.